### PR TITLE
A couple of bugfixes, a testcase and some formatting

### PR DESCRIPTION
**Pull request by SeeSpotRun**
In other words, a small tidy-up for #245

### CODE REVIEW

File: lib/utilities.c [hunk 1]
```diff
@@ -1216,8 +1212,8 @@ RmOff rm_offset_get_from_path(_UNUSED const char *path, _UNUSED RmOff file_offse
     return 0;
 }
 
-int rm_offsets_match(char *path1, char *path2) {
-    return (path1 == path2);
+RmOffsetsMatchCode rm_offsets_match(char *path1, char *path2) {
+    return (path1 == path2) ? RM_OFFSETS_MATCH : RM_OFFSETS_NO_DATA;
```
Comment (sahib) [inline lib/utilities.c]: Why do we compare here by pointer again?

File: tests/test_mains/test_dedupe.py [hunk 2]
```diff
@@ -160,10 +162,66 @@ def test_dedupe_works():
...
             '--is-reflink', path_a, path_b,
             use_default_dir=False,
             with_json=False,
             verbosity=""
         )
+
+# count the number of line in a file which start with patterns[]
+def pattern_count(path, patterns):
+    counts = [0] * len(patterns)
+    f = open(path, 'r')
```
Comment (sahib) [inline tests/test_mains/test_dedupe.py]: ``with open`` is your friend. :) (Sorry, too much python lately...)

File: lib/utilities.c (hunk 1 above)
Comment (SeeSpotRun) [inline lib/utilities.c]: [embarrassing](https://github.com/sahib/rmlint/commit/cf89f219eb5de0fd74104848a48924d0652f1fdb) legacy of my non-c background

Will fix... and maybe rename `rm_offsets_match` to `rm_link_type(path1, path2)` which will help avoid confusion

File: tests/test_mains/test_dedupe.py (hunk 2 above)
Comment (SeeSpotRun) [inline tests/test_mains/test_dedupe.py]: python wisdom gladly accepted

**Comment by SeeSpotRun** [comment]
Incorporated your feedback thanks.

Also found another [bug](https://github.com/sahib/rmlint/pull/247/commits/5758330840b51557dcaa0aee6ee7c3a0009ba6b1) along the way.  This would have reported files on different filesystems as reflinks, if they had the same physical offsets.

**Comment by sahib** [comment]
Merged, thanks.

**Pull request merged**